### ISSUE TITLE
Unable to use debug mode SDK49

### ISSUE THREAD

**Issue by yonitou** [expo/expo#24425]
### Minimal reproducible example

I don't have one since I'm using Gitlab

### Summary

Hi, 

I'm not able to use the debug mode of SDK 49 in my dev client build. 

When I'm pressing "Open JS Debugger", nothing happens. 

If I'm pressing "J" key in my terminal, the server is crashing with the following stack trace : 

```
FetchError: invalid json response body at http://192.168.1.81:8081/json/list reason: Unexpected token < in JSON at position 0
FetchError: invalid json response body at http://192.168.1.81:8081/json/list reason: Unexpected token < in JSON at position 0
    at /Users/yonitouboul/code/yonitou/hygo/node_modules/node-fetch/lib/index.js:273:32
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async Object.queryAllInspectorAppsAsync (/Users/yonitouboul/code/yonitou/hygo/node_modules/@expo/dev-server/build/JsInspector.js:53:30)
    at async DevServerManagerActions.openJsInspectorAsync (/Users/yonitouboul/code/yonitou/hygo/node_modules/@expo/cli/build/src/start/interface/interactiveActions.js:94:22)
    at async KeyPressHandler.handleKeypress (/Users/yonitouboul/code/yonitou/hygo/node_modules/@expo/cli/build/src/start/interface/KeyPressHandler.js:43:17)
```

Any idea ? 

### Environment
```
expo-env-info 1.0.5 environment info:
    System:
      OS: macOS 13.5.2
      Shell: 5.9 - /bin/zsh
    Binaries:
      Node: 18.12.1 - ~/.nvm/versions/node/v18.12.1/bin/node
      npm: 9.6.7 - ~/.nvm/versions/node/v18.12.1/bin/npm
    Managers:
      CocoaPods: 1.11.2 - /Users/yonitouboul/.rbenv/shims/pod
    SDKs:
      iOS SDK:
        Platforms: DriverKit 21.4, iOS 16.0, macOS 12.3, tvOS 16.0, watchOS 9.0
    IDEs:
      Android Studio: 2022.3 AI-223.8836.35.2231.10671973
      Xcode: 14.0.1/14A400 - /usr/bin/xcodebuild
    npmPackages:
      @expo/metro-config: 0.10.7 => 0.10.7 
      @expo/webpack-config: ^19.0.0 => 19.0.0 
      babel-preset-expo: 9.5.2 => 9.5.2 
      expo: 49.0.10 => 49.0.10 
      react: 18.2.0 => 18.2.0 
      react-dom: 18.2.0 => 18.2.0 
      react-native: 0.72.4 => 0.72.4 
    npmGlobalPackages:
      eas-cli: 5.2.0
      expo-cli: 6.3.2
    Expo Workflow: managed
```

**Comment by senghuotlay** [comment]
I am also having the same issue, were u able to solve it?

**Comment by dhcmega** [comment]
Since we moved to SDK49 debugging is hell.

Hot auto reload doesn’t work properly, not always refreshes the code., and no error/warning is given, so we end up troubleshooting with outdated code without knowing, so we are having to hard reload after every change.

We are unable to visualize redux store, including state history, as we did previously with React Native Debugger.

Unable to run devTools at the same time as VSCode breakpoints. When VSCode breakpoints do run, the experience is painstakingly slow and unresponsive.

Network tab randomly does not capture request and require restarting the debugger.

I think that would be all our major problems.Thanks

The debugger disconnects with an error message like "debugger disconnects for no reason".

On SDK48 there are no problems.

Thanks!
